File: Alcoholtracker/Services/JamCodeGenerator.swift
import Foundation

enum JamCodeGenerator {
    static let codeLength = 6
    static func generate() -> String {
        let chars = Array("ABCDEFGHJKLMNPQRSTUVWXYZ23456789")
        return String((0..<codeLength).map { _ in chars.randomElement() ?? "A" })
    }
}

File: android/app/src/main/kotlin/de/tipau/promille/service/JamCodeGenerator.kt
package de.tipau.promille.service

object JamCodeGenerator {
    const val CODE_LENGTH = 6
    private const val CHARACTERS = "ABCDEFGHJKLMNPQRSTUVWXYZ23456789"

    fun generate(): String = (1..CODE_LENGTH).map { CHARACTERS.random() }.joinToString("")

    fun sanitize(input: String): String =
        input.uppercase().filter { it.isLetter() || it.isDigit() }.take(CODE_LENGTH)
}

